src/org/parosproxy/paros/core/scanner/VariantJSONQuery.java

```diff
@@ -189,8 +190,11 @@ private void parseValue(String fieldName) {
         // Check if the value is a string        
         if (chr == QUOTATION_MARK) {
             int beginToken = sr.getPosition();
-            while ((chr = sr.read()) != QUOTATION_MARK) {
-                if (chr == -1) {
+            while (true) {

```

Comment:
Worth extracting a method and change the read of the property name to also use it (to fixing that one as well).